### ISSUE TITLE
Will it be made into a more comprehensive framework later on? 

### ISSUE THREAD

**Issue by HHC26** [gin-gonic/gin#3847]
link: https://github.com/gogf/gf

**Comment by guava-coder** [comment]
As you said, every developer has their own style of directory for their favorite projects. Will you going to make every style in one framework?

Frameworks like Spring Boot and Laravel that use MVC design patterns and provide project generation. However, you need to understand MVC, and how the project configuration works. Do you need MVC and project generation whenever use gin?

I prefer gin to remain simple, you can start a project with a few .go and .mod files, and it is ok. Gin doesn't give you a unified convention, but you can decide at any time.

**Comment by HHC26** [comment]
> As you said, every developer has their own style of directory for their favorite projects. Will you going to make every style in one framework?
> 
> Frameworks like Spring Boot and Laravel that use MVC design patterns and provide project generation. However, you need to understand MVC, and how the project configuration works. Do you need MVC and project generation whenever use gin?
> 
> I prefer gin to remain simple, you can start a project with a few .go and .mod files, and it is ok. Gin doesn't give you a unified convention, but you can decide at any time.

I also agree with what you said ，but I also believe that these features are quite good,
eg:
1. automatic codes generating for efficiency
2. robust engineering design specifications
3. convenient development CLI tool provide
4. openAPIV3 documentation generating, automatically

**Comment by guava-coder** [comment]
> convenient development CLI tool provide

Gin has no official CLI tool, but the community makes several of them.
Here are some Gin CLI tools I found via Google search:

- [nunu](https://github.com/go-nunu/nunu)
- [gin-cli](https://github.com/G-Node/gin-cli)
- [gin-annotation](https://github.com/1-st/gin-annotation)

> openAPIV3 documentation generating, automatically

I think you're talking about [gin-swagger](https://github.com/swaggo/gin-swagger)

**Comment by guava-coder** [comment]
> automatic codes generating for efficiency

If some codes are regularly used, you can write snippets for it. Most IDEs and editors can store snippets. However, I will avoid snippets and copy-paste on backend code, to prevent bugs and bad performance issues.
There are many different ways to increase efficiency while writing Go,  here are some tips:

1. Writing Unit test with testing package.
2. Using Generic to make reusable components

example:
```
    type Data interface {
	User
    }

    type User struct {
	Id       string
	Name     string
	/*other fields...*/
    }

    func ReadAndHandleRequestBody[T Data](ctx *gin.Context, operation func(T)) {
         /*your code...*/
    }
```
3. Using APIs provided by Gin

For example, you have `*gin.Context.JSON( )` and `gin.H{ }` to handle response:

```
func (serv UserService) QueryById(ctx *gin.Context, id string) {
	...
        if err == nil {
		ctx.JSON(http.StatusOK, gin.H{
			"Response": "Found User",
			"User":     res,
		})
	} else {
		ctx.JSON(http.StatusBadRequest, gin.H{
			"Response": "User not found",
		})
	}
}
```

**Comment by HHC26** [comment]
> I think you're talking about [gin-swagger](https://github.com/swaggo/gin-swagger)

The gin swagger is not easy to use, if Gin can call it, it will be easier to generate API documentation
